test/development/typescript-plugin/metadata/warn-no-type.test.ts

```diff
@@ -0,0 +1,134 @@
+import type { PluginLanguageService } from '../test-utils'
+import {
+  getPluginLanguageService,
+  getTsFiles,
+  NEXT_TS_ERRORS,
+} from '../test-utils'
+import { join } from 'path'
+
+const fixturesDir = join(__dirname, 'app/warn-no-type')
+
+describe('typescript-plugin - metadata - warn-no-type', () => {
+  let languageService: PluginLanguageService
+
+  beforeAll(() => {
+    languageService = getPluginLanguageService(__dirname)
+  })
+
+  it('should not have diagnostics for metadata with type', () => {
+    const hasTypeDir = join(fixturesDir, 'metadata', 'has-type')
+    const tsFiles = getTsFiles(hasTypeDir)
+    expect(tsFiles.length).toBe(8)
```

Comment:
Why is the number of files relevant? By checking a number you wouldn't know if during a change a file got added while another one was removed. It should be obvious from the test which file has which diagnostics.